torch/_dynamo/output_graph.py

```diff
@@ -540,15 +528,27 @@ def call_user_compiler(self, gm: fx.GraphModule) -> CompiledFn:
             _step_logger()(logging.INFO, f"calling compiler function {name}")
             compiler_fn = self.compiler_fn
             if config.verify_correctness:
-                compiler_fn = wrap_compiler_fn(compiler_fn)
-            compiled_fn = compiler_fn(gm, self.example_inputs())
+                compiler_fn = WrapperBackend(compiler_fn, self.example_inputs())
+            compiled_fn = compiler_fn(gm, self.fake_example_inputs())
             _step_logger()(logging.INFO, f"done compiler function {name}")
             assert callable(compiled_fn), "compiler_fn did not return callable"
         except Exception as e:
             compiled_fn = gm.forward
             raise BackendCompilerFailed(self.compiler_fn, e) from e
         return compiled_fn
 
+    def fake_example_inputs(self) -> List[torch.Tensor]:
+        result = []
+        for arg in self.graphargs:
+            example = arg.get_fake_examples()
+            if example is not None:
+                result.extend(example)
+            else:
+                # Fallback, in case fake_tensor was not set
+                # Particularly for graph args that are not tensors
+                result.extend(arg.get_examples())
```

Comment:
Two questions about this comment:

(1) The return type of this func is `List[torch.Tensor]`, is that wrong? (comment implies we have to worry about non-tensor graph args)

(2) Should this function maintain the invariant that it never returns real tensors, and assert here that `get_examples()` in the fallback only returns non-tensors?